### DISCUSSION TITLE
Lock annots

### DISCUSSION THREAD

**Discussion by Mano2731995** [pymupdf/PyMuPDF#2060] [Looking for help]
Hai jorj
Is any possible to lock the annotations? 

For example I have added pushpin button by using add_fileannot.
By using the mouse click the button and drag  anywhere in pdf file but need to lock the movement.

**Comment by JorjMcKie** [comment]
There is a flag bit for this - can't remember the name right now, but it is documented.

**Comment by Mano2731995** [comment]
> There is a flag bit for this - can't remember the name right now, but it is documented.

Ya jorj i got it from documents but proper example not available in document. Kindly share syntax for annot is locked
![IMG_20221118_213308.jpg](https://user-images.githubusercontent.com/117537492/202749026-5d5ca72a-d693-4dab-9e8c-59ff9569d7ec.jpg)

**Comment by JorjMcKie** [comment]
Well, this works as always of course: get the current flags value from the annot, then OR the desired bit into it via the `|` operator and then then the flags again in the annot.
Nothing specific here just normal Python.

**Reply by Mano2731995** [reply]
Hai jorj 
I got output thanks and other requirement is if any possible to change the annotation button size. I need small size of pushpin button instead of current size. 

 ![Screenshot_2022-11-18-14-17-25-148_com.android.chrome.jpg](https://user-images.githubusercontent.com/117537492/202838883-5e2ddaf3-2298-47e4-9f90-36dd805955d6.jpg)

**Comment by JorjMcKie** [comment]
Change the annot rectangle, `set_rect()` changes size and position on page.

**Reply by Mano2731995** [reply]
Position changed. How will do change the rect size? ![IMG_20221119_151556.jpg](https://user-images.githubusercontent.com/117537492/202845008-6ba141a1-8024-4d13-978b-90f4a0efc35b.jpg)

**Reply by JorjMcKie** [reply]
Well, use an appropriate rectangle, but **_do not_** `update()` **_the annot after_** `set_rect()`, because this will rebuild the standard, which is a 20 x 20 rectangle.
Make all your changes, then update, then `set_rect()`.

**Reply by JorjMcKie** [reply]
Example:
```python
a=page.add_file_annot((100,100),buffer,filename="test.pdf")
a.rect
Rect(100.0, 100.0, 120.0, 120.0)
a.set_rect(a.rect+(0,0,80,80))
a.rect
Rect(100.0, 100.0, 200.0, 200.0)
a.update()  # resizes to a 20x20 rect!
a.rect
Rect(141.0, 141.0, 159.0, 159.0)
```

**Reply by Mano2731995** [reply]
> Example:
> ```python
> a=page.add_file_annot((100,100),buffer,filename="test.pdf")
> a.rect
> Rect(100.0, 100.0, 120.0, 120.0)
> a.set_rect(a.rect+(0,0,80,80))
> a.rect
> Rect(100.0, 100.0, 200.0, 200.0)
> a.update()  # resizes to a 20x20 rect!
> a.rect
> Rect(141.0, 141.0, 159.0, 159.0)
> ```

I gave a.set_rect(a.rect+(0,0,1,1))  button size not changed. when I execute a.set_rect(a.rect+(0,0,80,80)) button size was ![IMG_20221119_173714.jpg](https://user-images.githubusercontent.com/117537492/202850961-3d61be2d-b102-401a-b60d-b8a5de13cbc4.jpg)changed but I need very smaller size of button ![IMG_20221119_173803.jpg](https://user-images.githubusercontent.com/117537492/202850933-c78a8f13-dd9c-4da5-b07d-2b3a8e285b74.jpg)

**Reply by JorjMcKie** [reply]
Aha, but what is your question?
You know that the standard button size is 20x20, so to get a 10x10 button do `a.set_rect(a.rect-(0,0,10,10))`.

**Reply by Mano2731995** [reply]
> Aha, but what is your question?
> You know that the standard button size is 20x20, so to get a 10x10 button do `a.set_rect(a.rect-(0,0,10,10))`

Thank u jorj. My mistake only I forgot to change - 
I got output thanks

**Reply by Mano2731995** [reply]
> > Aha, but what is your question?
> > You know that the standard button size is 20x20, so to get a 10x10 button do `a.set_rect(a.rect-(0,0,10,10))`
> 
> Thank u jorj. My mistake only I forgot to change - 
> I got output thanks

I got it jorj thanks for your quick support
